File: shared/types/analysis.ts
export interface Analysis {
  id: string;
  user_id: string;
  original_text: string;
  summary: string;
  keywords: string[];
  sentiment: SentimentResult;
  created_at: string;
  updated_at: string;
}

export interface SentimentResult {
  label: 'positive' | 'negative' | 'neutral';
  confidence: number;
}

export interface AnalysisRequest {
  text: string;
}

export interface AnalysisResponse {
  id: string;
  summary: string;
  keywords: string[];
  sentiment: SentimentResult;
  created_at: string;
}

export interface AnalysisHistory {
  analyses: Analysis[];
  total: number;
  page: number;
  limit: number;
}

File: backend/api/analysis.py
from fastapi import APIRouter, HTTPException, Depends
from pydantic import BaseModel
from typing import List, Optional

router = APIRouter()

class AnalysisRequest(BaseModel):
    text: str

class SentimentResult(BaseModel):
    label: str  # 'positive', 'negative', 'neutral'
    confidence: float

class AnalysisResponse(BaseModel):
    id: str
    summary: str
    keywords: List[str]
    sentiment: SentimentResult
    created_at: str

class AnalysisHistory(BaseModel):
    analyses: List[AnalysisResponse]
    total: int
    page: int
    limit: int

@router.post("/analyze", response_model=AnalysisResponse)
async def analyze_text(request: AnalysisRequest):
    """
    Analizar texto usando Gemini Pro
    """
    # TODO: Implementar análisis con Gemini Pro
    # TODO: Guardar resultado en Supabase
    
    return {
        "id": "fake_analysis_id",
        "summary": f"Resumen del texto: {request.text[:50]}...",
        "keywords": ["palabra1", "palabra2", "palabra3"],
        "sentiment": {
            "label": "positive",
            "confidence": 0.85
        },
        "created_at": "2024-01-01T00:00:00Z"
    }

@router.get("/history", response_model=AnalysisHistory)
async def get_analysis_history(
    page: int = 1,
    limit: int = 10
):
    """
    Obtener historial de análisis del usuario
    """
    # TODO: Implementar consulta a Supabase
    # TODO: Implementar autenticación de usuario
    
    return {
        "analyses": [],
        "total": 0,
        "page": page,
        "limit": limit
    }

@router.get("/history/{analysis_id}", response_model=AnalysisResponse)
async def get_analysis_by_id(analysis_id: str):
    """
    Obtener análisis específico por ID
    """
    # TODO: Implementar consulta a Supabase
    # TODO: Verificar que el análisis pertenece al usuario autenticado
    
    raise HTTPException(status_code=404, detail="Analysis not found")